### DISCUSSION TITLE
Node Library?

### DISCUSSION THREAD

**Discussion by melMass** [nxt-dev/nxt_editor#229] [Q&A] [answered]
Hi,

I discovered nxt 2 days ago and I have been using it a lot since! I really like the paradigm.

One thing I find a bit cumbersome is building a node library, and I wanted to know the best approach.
I've seen all the videos about nxt as well and @ImLucasBrown mentioned that this was being reworked.

Here is my approach right now, please any feedback is welcome:

Let's take a simple filesystem/OS node library, I have a few nodes like `list_folder`, `mkHierarchy`, `writeJSONtoFile` ...
Each of them are a separated nxt graph and most return values using the `STAGE` object.
I then created a `FS Subgraphs` layer/graph in which I instantiate all of them. This is then the library I import into a layer on my other projects.

This works quite well but I feel like it's a bit cumbersome to update/initiate.
I might think of this whole thing wrong as I still discover new things while experimenting.

Thanks

**Comment by ImLucasBrown** [comment, marked as answer]
More or less that is how we and other's use NXT to organize "library" nodes.
Referencing various graphs containing "utility" nodes into one is a good solution. It allows you to update the underlying nodes and version control them without touching any graphs that _actually_ use those nodes. We've seen users manage workflows and "library" graphs like this with great success. Also, with the recent addition of layer locks, it's a lot easier to avoid accidental edits of those referenced layers.

Some day we'd like to have a more robust system for managing builtin/library nodes, and perhaps even a way to share those collections with the community.
